lib/ZwaveClient.js

```diff
@@ -702,7 +703,23 @@ function updateValueMetadata (zwaveNode, zwaveValue, zwaveValueMeta) {
  * @param { ZWaveNode } zwaveNode
  * @param { ValueAddedArgs } valueAddedArgs https://github.com/zwave-js/node-zwave-js/blob/cb35157da5e95f970447a67cbb2792e364b9d1e1/packages/core/src/values/ValueDB.ts#L8
  */
-function addValue (zwaveNode, zwaveValue, discovery) {
+
+function discoverDevice (zwaveNode, zwaveValue) {
+  const node = this.nodes[zwaveNode.id]
+
+  if (!node) {
+    debug('discoverDevice: no such node: ' + zwaveNode.id, 'error')
+  } else {
+    const zwaveValueMeta = zwaveNode.getValueMetadata(zwaveValue)
+
+    const valueId = initValue.call(this, zwaveNode, zwaveValue, zwaveValueMeta)
+
+    debug(`Node ${zwaveNode.id}: hass Discovery for ${valueId.id}`)
+
+    this.emit('readyForDiscovery', valueId, node)
+  }
+}
```

Comment:
@billiaz Actually this is almost the same as `addValue` function. I suggest to add a bool flag instead to `addValue` and also to the event triggered

Reply:
I am finishing some changes! and I will post a new function, smaller, simpler